internal/execution/execution_test.go

```diff
@@ -345,3 +348,113 @@ func Test_CallTarget(t *testing.T) {
 		})
 	}
 }
+
+func Test_handleResponse(t *testing.T) {
+	type args struct {
+		resp *http.Response
+	}
+	type res struct {
+		data    []byte
+		wantErr func(error) bool
+	}
+	tests := []struct {
+		name string
+		args args
+		res  res
+	}{
+		{
+			"response, statuscode unknown and body",
+			args{
+				resp: &http.Response{
+					StatusCode: 1000,
+					Body:       io.NopCloser(bytes.NewReader([]byte("body"))),
+				},
+			},
+			res{
+				wantErr: func(err error) bool {
+					if errors.Unwrap(errors.Unwrap(errors.Unwrap(err))).Error() != "body" {
```

Comment:
This look too wild to provide a nice UX for clients

Reply:
that was necessary as the errors.New doesn't provide the errros.Is function to check if they are in the chain, so unwrap to compare the string